### PR TITLE
Feature/find tag

### PR DESCRIPTION
**Pull request by manuellagrand**
this pull request include the temporary traduction of each layout and the implementation of tag search form with auto-suggestion

### CODE REVIEW

File: src/common/services/translator.ts [hunk 1]
```diff
@@ -28,8 +28,8 @@ const initI18n = i18n.init({
 
 export enum AvailableLanguages {
     en = "English",
-    fr = "Français",
-    de = "Deutch",
+    fr = "French",
```
Comment (danielweck) [inline src/common/services/translator.ts]: What is the rationale for changing from French to English here?

Comment (manuellagrand) [reply src/common/services/translator.ts]: the reason is to standardize languages in english which is the default language.

Comment (manuellagrand) [reply src/common/services/translator.ts]: i made some changes in translation files in order to translate each languages when it is selected in settings

**Comment by danielweck** [comment]
Let's say that I am a Japanese user, the UI is in English which is the default locale when I install the app. Now I need to find the menu / options dialog where I can switch to Japanese. I follow a tutorial to reach the UI (I cannot read English fluently so I follow steps / instructions). When I get there, I expect to see my language in my own script / locale, not English. In my opinion. What do you think?

**Comment by llemeurfr** [comment]
I understand your concern @danielweck. When I change the language in e.g. an iphone, I see both informations, ex. "Espanõl" (target language) + (below) "Spanish" or "Espagnol" (i.e. the current language). We'll work on obtaining that.

File: src/renderer/components/catalog/GridView.tsx [hunk 2]
```diff
@@ -52,8 +51,16 @@ export default class GridView extends React.Component<GridViewProps, undefined>
                             </section>
                         );
                 })}
-                <AddEntryForm />
+            <button className={styles.tag_add_button}>
+                <span>
+                    <Link to="/settings/tags" style={{color: "#1a1a1a"}}>
```
Comment (clebeaupin) [inline src/renderer/components/catalog/GridView.tsx]: Do not put inline styles

File: src/renderer/components/catalog/AddEntryForm.tsx [hunk 3]
```diff
@@ -7,86 +7,112 @@
...
+        console.log("tag: " + this.state.tag);
+    }
     private submit(e: any) {
+
+        if (this.state.tag === "") {
+            alert("tag input is empty.");
+            return;
+        }
         e.preventDefault();
+        console.log("props" + this.props);
```
Comment (clebeaupin) [inline src/renderer/components/catalog/AddEntryForm.tsx]: Remove all logs
Use log for debugging but do not commit them

File: src/renderer/components/catalog/AddEntryForm.tsx [hunk 4]
```diff
@@ -104,12 +130,19 @@ export default withApi(
...
-                moduleId: "publication",
-                methodId: "updateTags",
+                moduleId: "catalog",
+                methodId: "updateEntries",
             },
         ],
     },
 );
+
+export const translate = withTranslator(AddEntryForm);
```
Comment (clebeaupin) [inline src/renderer/components/catalog/AddEntryForm.tsx]: translator is already included in withApi so you don't need it to use it

File: src/renderer/components/catalog/AddEntryForm.tsx [hunk 5]
```diff
@@ -7,86 +7,112 @@
...
+    addEntry?: (entry: any) => void;
+    getentry?: CatalogEntryView[];
+    tags?: string[];
 }
 
 export class AddEntryForm extends React.Component<AddEntryFormProps, AddEntryFormState> {
     private selectRef: any;
+    // How to translate temporarily
+    @lazyInject("translator")
+    private translator: Translator;
```
Comment (clebeaupin) [inline src/renderer/components/catalog/AddEntryForm.tsx]: Stop using lazy inject.
Instead use withApi or withTranslator HOC

File: src/renderer/components/catalog/ListView.tsx [hunk 6]
```diff
@@ -12,24 +12,32 @@ import { CatalogEntryView } from "readium-desktop/common/views/catalog";
...
 import PublicationListElement from "readium-desktop/renderer/components/publication/PublicationListElement";
 
-import AddEntryForm from "./AddEntryForm";
+import { Link } from "react-router-dom";
 
 import * as styles from "readium-desktop/renderer/assets/styles/myBooks.css";
+import { TranslatorProps, withTranslator } from "../utils/translator";
 
-interface ListViewProps {
+import { lazyInject } from "readium-desktop/renderer/di";
```
Comment (clebeaupin) [inline src/renderer/components/catalog/ListView.tsx]: Remove lazyInject if it used to autowired Translator

**Pull request closed without merging**